globalnoc-tsds-datasource/src/datasource.js

```diff
@@ -449,15 +466,24 @@ export class GenericDatasource {
 
             var aggregation = 'aggregate(values.' + target.metricValues_array[index];
             aggregation += ', $quantify, ';
-
+			let template_variables = getVariableDetails();
 			if (target.aggregator[index] == "percentile") aggregation += target.aggregator[index]+'('+target.percentileValue[index]+'))';
-			else aggregation += target.aggregator[index]+')';
-
+			else if(target.aggregator[index] === "template") {
+
+				//console.log(target.aggregator[index]+ ": " + target.templateVariableValue[index]);	
+				//_.forEach(template_variables, (value, key) => console.log(`Key: ${key}, Value: ${value}`));
+				let template = target.templateVariableValue[index];
+        			aggregation += template_variables[template.slice(1,template.length)]+')';
+	    		} else aggregation += target.aggregator[index]+')';
+	  
             if (typeof target.metricValueAliases[index] === 'undefined' || target.metricValueAliases[index] === null) {
               target.metricValueAliases[index] = '';
             }
-            target.metricValueAliasMappings[aggregation.toString()] = target.metricValueAliases[index];
-
+	    let alias_var = target.metricValueAliases[index];
+	    let alias = template_variables[alias_var.slice(1, alias_var.length)];
```

Comment:
This might not actually exist. While the "template" metric will always require a value, the "alias" might not be a reference to a template variable. For example, I might just alias something as "Data" which doesn't map to any template variables. Trying that out on your instance results in no alias being used at all, presumably because finds no match in the template_variables object.